### PR TITLE
Debt/sl 995/sonar qube

### PR DESCRIPTION
**Pull request by leszekgu**

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/exceptions/OptimisticLockException.java [hunk 1]
```diff
@@ -1,12 +1,10 @@
 package uk.gov.hmcts.reform.sandl.snlapi.exceptions;
 
+import lombok.NoArgsConstructor;
 import org.springframework.http.HttpStatus;
 import org.springframework.web.bind.annotation.ResponseStatus;
 
 @ResponseStatus(value = HttpStatus.CONFLICT, reason = "The object version has changed")
+@NoArgsConstructor
 public class OptimisticLockException extends RuntimeException {
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlapi/exceptions/OptimisticLockException.java]: is this implementation following the best practices for writing custom exceptions in java?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtAuthenticationFilter.java [hunk 2]
```diff
@@ -18,7 +18,7 @@
 
 public class JwtAuthenticationFilter extends OncePerRequestFilter {
 
-    private static final Logger logger = LoggerFactory.getLogger(JwtAuthenticationFilter.class);
+    private static final Logger localLogger = LoggerFactory.getLogger(JwtAuthenticationFilter.class);
```
Comment (kpakur) [inline src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtAuthenticationFilter.java]: what is the common convention used in other classes across all projects here?

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/exceptions/OptimisticLockException.java (hunk 1 above)
Comment (leszekgu) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlapi/exceptions/OptimisticLockException.java]: https://stackify.com/java-custom-exceptions/
I will add documentation, which will also explain why there is only one constructor - shortly spring usage of @ResponseStatus and yes this is tightly coupled to spring.

File: src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtAuthenticationFilter.java (hunk 2 above)
Comment (leszekgu) [reply src/main/java/uk/gov/hmcts/reform/sandl/snlapi/security/JwtAuthenticationFilter.java]: Can't find any, the thing here is that SonarQube noticed we are extending a class which already have a field named `logger` and it's of a different type and it's a code smell.
As we want to log with proper class info I prefer to change field name here.

**Pull request merged**